### PR TITLE
Add ceres software

### LINKED ISSUE #217: Add CERES
Hook FitBenchmarking up to Google's CERES solver:
http://ceres-solver.org/

### PR DESCRIPTION
**Pull request by joshuadkitenge**
#### Description of Work
Fixes #217

Add ceres solver into fitbenchmarking 

- [x] Test that the ceres solver works with the PyCeres python bindings
- [x] Create ceres controller 
- [x] Test ceres controller on NIST problems and resolve any errors      

#### Testing Instructions

1. Test against different benchmark problems  
2.
3.

Function: Does the change do what it's supposed to?

Tests: Does it pass? Is there adequate coverage for new code?

Style: Is the coding style consistent? Is anything overly confusing?

Documentation: Is there a suitable change to documentation for this change?

### CODE REVIEW

**Comment by joshuadkitenge** [comment]
https://notes.andrewtorgesen.com/doku.php?id=public:ceres

**Comment by joshuadkitenge** [comment]
@AndrewLister-STFC 
**[ACTION]** Build docker container for ceres to fix failing test

File: docs/source/users/options/minimizer_option.rst [hunk 1]
```diff
@@ -64,6 +64,50 @@ The Bumps minimizers are set as follows:
...
+* `Levenberg-Marquardt <http://ceres-solver.org/nnls_solving.html#levenberg-marquardt>`__ (:code:`Levenberg-Marquardt`)
+* `Dogleg <http://ceres-solver.org/nnls_solving.html#dogleg>`__ (:code:`Dogleg`)
+* `Steepest Descent <http://ceres-solver.org/nnls_solving.html#line-search-methods>`__ (:code:`steepest_descent`)
+* `BFGS algorithm <http://ceres-solver.org/nnls_solving.html#line-search-methods>`__ (:code:`BFGS`)
+* `LBFGS algorithm <http://ceres-solver.org/nnls_solving.html#line-search-methods>`__ (:code:`LBFGS`)
+* `Fletcher-Reeves Non Linear Conjugate-Gradient <http://ceres-solver.org/nnls_solving.html#line-search-methods>`__ (:code:`Fletcher_Reeves`)
+* `Polak-Ribiere Non Linear Conjugate-Gradient <http://ceres-solver.org/nnls_solving.html#line-search-methods>`__ (:code:`Polak_Ribiere`)
+* `Hestenes-Stiefel Non Linear Conjugate-Gradient <http://ceres-solver.org/nnls_solving.html#line-search-methods>`__ (:code:`Hestenes_Stiefel`)
+
+**Licence** Ceres Solver is available `here <http://ceres-solver.org/license.html>`__
```
Comment (tyronerees) [inline docs/source/users/options/minimizer_option.rst:85]:
```suggestion
**Licence** Ceres Solver is available under the new BSD licence -- details can be found `here <http://ceres-solver.org/license.html>`__ 
```

**Review by tyronerees — CHANGES_REQUESTED** [review]
I've run locally on a couple of different cost functions and data sets, and all seems to work for me.

There's one minor documentation wording suggestion I've made, and one of the unit tests seems to be failing. 

Thanks for this -- it's great to have CERES here in FitBenchmarking 😄

**Pull request merged**